sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/scripting/LongJavaScriptCheck.java

```diff
@@ -28,39 +28,46 @@ public class LongJavaScriptCheck extends AbstractPageCheck {
 
   private static final int DEFAULT_MAX_LINES = 5;
 
+  private final StringBuilder text = new StringBuilder();
+
+  private TagNode scriptNode;
+
+  private int startingNodeLinePosition;
+
   @RuleProperty(
     key = "maxLines",
     description = "Max Lines (Number)",
     defaultValue = "" + DEFAULT_MAX_LINES)
   public int maxLines = DEFAULT_MAX_LINES;
 
-  private int linesOfCode;
-  private TagNode scriptNode;
+  @Override
+  public void startElement(TagNode node) {
+    if ("script".equalsIgnoreCase(node.getNodeName())) {
+      scriptNode = node;
+      startingNodeLinePosition = node.getStartLinePosition();
+      text.delete(0, text.length());
+    }
+  }
 
   @Override
   public void characters(TextNode textNode) {
-    if (scriptNode != null) {
-      linesOfCode += textNode.getLinesOfCode();
-
-      if (linesOfCode > maxLines) {
-        createViolation(scriptNode.getStartLinePosition(),
-          "The length of this JS script (" + linesOfCode + ") exceeds the maximum set to " + maxLines + ".",
-          Double.valueOf(linesOfCode) - Double.valueOf(maxLines));
-        scriptNode = null;
-      }
+    if(scriptNode != null){
```

Comment:
_minor_ this line as a weird styling
```suggestion
    if (scriptNode != null) {
```

Reply:
Great catch! Automatic reformating doesn't want to catch it weirdly.